### DISCUSSION TITLE
Support `--no-cache` Docker build flag

### DISCUSSION THREAD

**Discussion by wdhdev** [coollabsio/coolify#4401] [Feature Requests]
I'd like to have support for the `--no-cache` Docker build flag in Coolify as it is helpful if you need to specifically rebuild a specific image from scratch every time.

**Comment by andrasbacsai** [comment]
If you rebuild your application with `Force Deploy`, it basically adds the `--no-cache` docker option.

![image](https://github.com/user-attachments/assets/838b2d4b-dcb0-4651-9f7d-4534efc851fa)

**Reply by andrasbacsai** [reply]
I just reviewed this on stream and chat (thanks Yash) told me you would like to rebuild with no cache on every commit (I missread it).

Then this request makes sense. I will add the `--no-cache` option to the custom docker options in the next version.

**Reply by andrasbacsai** [reply]
I won't add it as a custom docker options, as custom docker options are related to the deployment part of your application not the build part.

It will be in the `Advanced` menu.

![image](https://github.com/user-attachments/assets/346cb7a0-5a5b-4c2e-851e-0e2807beef29)

**Reply by douglasg14b** [reply]
This doesn't appear to work. builds still consume cached layers, and stale build artifacts persists despite these settings....

Stale layers are making it effectively impossible to deploy changes. It's like a little bomb waiting to halt your stack.

Honestly at a loss.

**Reply by douglasg14b** [reply]
Potentially related to: https://github.com/coollabsio/coolify/issues/2895
